keps/sig-scheduling/5007-device-attach-before-pod-scheduled/README.md

```diff
@@ -0,0 +1,1030 @@
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+The basic idea is the following:
+
+1. **Add a Ready Flag to ResourceClaim**:
+   - Add a flag to `ResourceClaim` that indicates the readiness state of the device.
+   The `PreBind` phase will be held until this flag is set to "Ready".
+
+2. **Wait for Device Attachment Completion in the PreBind() Process**:
+   The overall flow of the PreBind() process is as follows:
+
+   - **Update ResourceClaim**:
+     - The scheduler updates the `ResourceClaim` to notify the vendor's driver that the device needs to be prepared.
+     This process is the same as the existing `PreBind`.
+     - After updating the `ResourceClaim`, if the flag is set to "Preparing", the completion of the `PreBind` phase will be held until the flag is set to "Ready".
+
+   - **Monitoring and Preparation by Composable DRA Controllers**:
+     - Composable DRA Controllers monitor the `ResourceClaim`.
+     If a device that requires preparation is associated with the `ResourceClaim`, they perform the necessary preparations.
+     Once the preparation is complete, they set the flag to "Ready".
+
+   - **Completion of the PreBind Phase**:
+     - Once the flag is set to "Ready", the `PreBind` phase is completed, and the scheduler proceeds to the next step.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### DRA Scheduler Plugin Design Overview
+
+This document outlines the design of the DRA Scheduler Plugin, focusing on the handling of fabric devices.
+Key additions include new attributes for device identification, enhancements to `AllocatedDeviceStatus`, and the process for handling `ResourceSlices` upon attachment failure.
+The composable controller design is also discussed, emphasizing efficient utilization of fabric devices.
+
+//TODO update figure
+![proposal](proposal2.JPG)
```

Comment:
This figure didn't show the timeout case (10 mins you mentioned in the following section).